.cirrus.star

```diff
@@ -1,4 +1,430 @@
+            "ARTIFACTORY_DEPLOY_USERNAME": "vault-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer",
+            "ARTIFACTORY_DEPLOY_PASSWORD": "VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer access_token]",
+            "ARTIFACTORY_DEPLOY_REPO": "sonarsource-public-qa",
+            "ARTIFACTORY_ACCESS_TOKEN": "VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-private-reader access_token]",
+            "CIRRUS_SHELL": "bash",
+            "CIRRUS_CLONE_DEPTH": 1,
+            "GRADLE_USER_HOME": "${CIRRUS_WORKING_DIR}/.gradle",
+            "GRADLE_COMMON_FLAGS": "--console plain --no-daemon --profile",
+            "GO_VERSION": "1.21.7",
+            "GO_CROSS_COMPILE": "1",
+            "PROTOC_VERSION": "25.0",
+            "ORG_GRADLE_PROJECT_signingKey": "VAULT[development/kv/data/sign data.key]",
+            "ORG_GRADLE_PROJECT_signingPassword": "VAULT[development/kv/data/sign data.passphrase]",
+            "ORG_GRADLE_PROJECT_signingKeyId": "0x7DCD4258",
+            "PROJECT_VERSION_CACHE_DIR": "project-version"
+        }
+    }
+
+
+def build_task():
+    return {
+        "build_task": {
+            "env": {
+                "SIGN_KEY": "VAULT[development/kv/data/sign data.key]",
+                "PGP_PASSPHRASE": "VAULT[development/kv/data/sign data.passphrase]",
+                "SONAR_HOST_URL": "https://sonarcloud.io",
+                "SONAR_TOKEN": "VAULT[development/kv/data/sonarcloud data.token]",
+                "DEPLOY_PULL_REQUEST": "true",
+                "BUILD_ARGUMENTS": "-DtrafficInspection=false --parallel --profile -x test -x sonar"
+            },
+            "eks_container": {
+                "dockerfile": ".cirrus/Dockerfile",
+                "docker_arguments": {
+                    "CIRRUS_AWS_ACCOUNT": "${CIRRUS_AWS_ACCOUNT}",
+                    "GO_VERSION": "${GO_VERSION}",
+                    "PROTOC_VERSION": "${PROTOC_VERSION}"
+                },
+                "cluster_name": "${CIRRUS_CLUSTER_NAME}",
+                "builder_role": "cirrus-builder",
+                "builder_image": "docker-builder-v*",
+                "builder_instance_type": "t3.small",
+                "builder_subnet_id": "${CIRRUS_AWS_SUBNET}",
+                "region": "eu-central-1",
+                "namespace": "default",
+                "use_in_memory_disk": "true",
+                "memory": "6G",
+                "cpu": 10
+            },
+            "project_version_cache": {
+                "folder": "$PROJECT_VERSION_CACHE_DIR",
+                "fingerprint_script": "echo $BUILD_NUMBER",
+                "populate_script": "mkdir -p \"$PROJECT_VERSION_CACHE_DIR\"\ntouch \"$PROJECT_VERSION_CACHE_DIR\"/evaluated_project_version.txt\n",
+                "reupload_on_changes": "true"
+            },
+            "create_gradle_directory_script": [
```

Comment:
This block is used 6 times in the whole file. Is it expected such duplication? I thought the template would have more building blocks and the duplication would be avoided.

Reply:
This is fixed in the next iteration.
The duplication helps to expose the common building blocks in an explicitly way, then the DRY phase can start.